src/currencyPage.tsx

```
@@ -1,28 +1,77 @@
-import { useEffect } from "react";
+import { useEffect, useState } from "react";
 import "./currencyPage.css";
+import { Currency, mockCurrencyAPI, Rates } from "./api/mockCurrencyAPI";
+import { Input, Select } from "antd";
 
-const Base_URL =
-  "https://cdn.jsdelivr.net/gh/fawazahmed0/currency-api@1/latest/currencies/eur/jpy.json";
 function CurrencyPage() {
+  const [currencies, setCurrencies] = useState<Partial<Rates>>({});
+  const [fromCurrency, setFromCurrency] = useState<Currency>("USD");
+  const [toCurreny, setToCurreny] = useState<Currency>("EUR");
+  const [fromAmount, setFromAmount] = useState(0);
+  const [toAmount, setToAmount] = useState(0);
+
   useEffect(() => {
-    fetch(Base_URL)
-      .then((res) => res.json)
-      .then((data) => console.log("err"));
+    mockCurrencyAPI().then((data: any) => {
+      setCurrencies(data);
+    });
   }, []);
+
+  function handleChange(e: React.ChangeEvent<HTMLInputElement>) {
+    e.preventDefault();
+    const numValue = Number(e.target.value);
+    setFromAmount(numValue);
+    setToAmount(
+      (fromAmount * currencies[fromCurrency]!) / currencies[toCurreny]!
+    );
+  }
+
   return (
     <>
-      <p>convert</p>
-      <div className="currenyPage">
-        <input type="number"></input>
-        <select>
-          <option value="hi"></option>
-        </select>
+      <br></br>
+      <form>
+        <Input
+          className="antd-input"
+          type="number"
+          value={fromAmount.toString()}
+          onChange={handleChange}
+          //onChange={(e) => setFromAmount(Number(e.target.value))}
+        ></Input>
+
+        <Select
+          value={fromCurrency}
+          // @ts-ignore
```

no need to ignore the error here, let's fix it instead